holidays/financial/ny_stock_exchange.py

```diff
@@ -73,8 +71,13 @@ def _populate(self, year):
         # REGULAR HOLIDAYS
         ##############################################################
         # NYD
-        nyd = date(year, JAN, 1)
-        self._set_observed_date(nyd, "New Year's Day", always_post=True)
+        this_year_nye = date(year, DEC, 31)
+        if this_year_nye.isoweekday() == 5:  # check if NYE should be observed
```

Comment:
This case deserves detailed explanation. I suggest combining these 2 comments into a single one above the if block.